include/maliput/routing/route_phase.h

```diff
@@ -0,0 +1,174 @@
+      const maliput::api::RoadPosition& start_road_position,
+      const maliput::api::RoadPosition& end_road_position,
+      const maliput::api::LaneSRange& default_lane_s_ranges,
+      const std::vector<maliput::api::LaneSRange>& adjancent_lane_s_ranges,
+      const maliput::api::RoadNetwork* road_network) :
+        index_(index), must_switch_(must_switch), lane_s_range_tolerance_(lane_s_range_tolerance),
+        default_lane_s_range_(default_lane_s_range), adjancent_lane_s_ranges_(adjancent_lane_s_ranges),
+        road_network_(road_network) {
+    MALIPUT_THROW_UNLESS(index_ >= 0);
+    MALIPUT_THROW_UNLESS(lane_s_range_tolerance >= 0.);
+    MALIPUT_THROW_UNLESS(road_network_ != nullptr);
+    MALIPUT_THROW_UNLESS(start_road_position_.lane != nullptr);
+    MALIPUT_THROW_UNLESS(end_road_position_.lane != nullptr);
+    // TODO(#543): Validate that start_road_position_.lane->id() and position are in the default_lane_s_range and
+    // adjacent_lane_s_ranges.
+    // TODO(#543): Validate that start_road_position_.lane->id() and position are in the default_lane_s_range and
+    // adjacent_lane_s_ranges.
+    // TODO(#543): Validate that adjancent_lane_s_ranges_ are effectively adjacent to default_lane_s_range.
+  }
+
+  /// @return The index of this RoutePhase.
+  int Index() const { return index_; }
+
+  /// @return The start maliput::api::RoadPosition of this RoutePhase.
+  const maliput::api::RoadPosition& StartRoadPosition() const { return start_road_position_; }
+
+  /// @return The end maliput::api::RoadPosition of this RoutePhase.
+  const maliput::api::RoadPosition& EndRoadPosition() const { return end_road_position_; }
+
+  /// @return The default maliput::api::LaneSRange.
+  const maliput::api::LaneSRange& DefaultLaneSRange() const { return default_lane_s_range_; }
+
+  /// @return The maliput::api::rules::RoadRulebook::QueryResults for the
+  /// default maliput::api::LaneSRange.
+  maliput::api::rules::RoadRulebook::QueryResults RulesForDefaultLaneSRange() const {
+    return road_network_->rulebook()->FindRules({default_lane_s_range_});
+  }
+
+  /// @return A vector of maliput::api::LaneSRanges with the adjacent
+  /// maliput::api::LaneSRanges.
+  std::vector<maliput::api::LaneSRange> AdjacentLaneSRanges() const { return adjancent_lane_s_ranges_; }
+
+  /// Finds the rules for @p lane_s_range.
+  ///
+  /// The @p lane_s_range must intersect one of the default or adjacent
+  /// maliput::api::LaneSRanges.
+  /// (To be discussed): we probably want it to be included rather than intersected.
+  ///
+  /// @param lane_s_range The maliput::api::LaneSRange to consider.
+  /// @return The rules that apply to that @p lane_s_range.
+  /// @throws maliput::common::assertion_error When @p lane_s_range does not
+  /// intersect any of the default or adjacent maliput::api::LaneSRanges.
+  maliput::api::rules::RoadRulebook::QueryResults RulesForLaneSRange(
+      const maliput::api::LaneSRange& lane_s_range) const;
```

Comment:
I wonder if this makes sense. The rule api is rather extensive. Rules, Rule states, Rule state currently applying. Rule Types. How do you decide up to which point cover the api?

 Having the LaneSRange should be sufficient to go directly to the Rule API that maliput provides and query what the agent needs.

Reply:
Before making any change to the code, I'd like to iterate a bit over this.
I think it's a fair point. Having proxies to other queries is complicated to maintain and leaves things out of the table. On the other side, it allows agent behavior to quickly accessing the data it requires to navigate.

I am not a big fan of these methods, but I definitely see the positive flip side.

Reply:
@liangfok @stonier wdyt?